app/src/main/java/com/asfoundation/wallet/topup/TopUpFragment.kt

```diff
@@ -184,6 +185,12 @@ class TopUpFragment : BasePageViewFragment(), TopUpFragmentView {
       adapter = topUpAdapter
     }
     view.viewTreeObserver.addOnGlobalLayoutListener(listener)
+    binding.offerWallMsg.setOnClickListener { ChallengeRewardManager.onNavigate() }
+  }
+
+  override fun onResume() {
```

Comment:
I recommend changing onResume to onCreate, onResume is called much more often than onCreate